### PR TITLE
Gate HSDP by torch 2.1.0

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Gate torch import by 2.1.0. We are not confident checkpointing works on earlier versions per @eracah

### CODE REVIEW

**Review by b-chu — COMMENTED** [review]
Do we know that it breaks on 2.0.1 and works on 2.1.0 for certain?

**Comment by mvpatel2000** [comment]
> Do we know that it breaks on 2.0.1 and works on 2.1.0 for certain?

We know for sure it works on 2.1.0. Community users reported issues with 2.0.1

File: tests/trainer/test_fsdp_checkpoint.py [hunk 1]
```diff
@@ -556,7 +556,8 @@ def test_fsdp_partitioned_state_dict_load(world_size, tmp_path: pathlib.Path, st
 @pytest.mark.parametrize('autoresume', [False, True])  # True commented out for now
 @pytest.mark.parametrize('num_shards', [2, 4, 7])
 @pytest.mark.parametrize('sharding_strategy', ['FULL_SHARD', 'SHARD_GRAD_OP'])
-@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.0.1'), reason='requires PyTorch 2.01 or higher')
+@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.0.1'),
```
Comment (j316chuck) [inline tests/trainer/test_fsdp_checkpoint.py:559]: should this be <=?

File: composer/utils/checkpoint.py [hunk 2]
```diff
@@ -313,7 +313,7 @@ def load_sharded_checkpoint(
             f'Sharded checkpoint loading requires torch version >= 2.0.0. You have torch version {torch.__version__}')
 
     using_multinode = dist.get_world_size() != dist.get_local_world_size()
-    if not using_torch_2_0_1() and using_multinode:
+    if not version.parse(torch.__version__) >= version.parse('2.0.1') and using_multinode:
```
Comment (j316chuck) [inline composer/utils/checkpoint.py:316]: should this be 2.1.0?

**Review by j316chuck — APPROVED** [review]
should we pin the version for 2.1.0?

File: tests/trainer/test_fsdp_checkpoint.py (hunk 1 above)
Comment (mvpatel2000) [reply tests/trainer/test_fsdp_checkpoint.py:559]: Nope, it is for 2.0. I just fixed a typo

File: composer/utils/checkpoint.py (hunk 2 above)
Comment (mvpatel2000) [reply composer/utils/checkpoint.py:316]: nope, just removing utility fn and only doing for 2.0 to be consistent

**Pull request merged**